tests/5.1/metadirective/test_metadirective_nothing.c

```diff
@@ -0,0 +1,98 @@
+//===--------------------- test_metadirective_nothing.c ---------------------===//
+//
+// OpenMP API Version 5.1 Nov 2020
+// 
+// Test for nothing directive within metadirectives. Runs a variety of
+// metadirectives that check if the nothing directive is properly rendered.
+// Primarily tests based on the fact that no matter what 'when' clause is 
+// rendered it should result in nothing, and thus no additional pragma should
+// be created. Thus, the threads should remain unchanged through this process
+// and the compiler should handle it properly. Handles both device and host
+// based tests.
+//
+////===---------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 1024
+
+int metadirectiveOnDevice() {
+   int errors = 0;
+   int A[N];
+
+   for (int i = 0; i < N; i++) {
+      A[i] = 0;
+   }
+
+   #pragma omp target map(tofrom: A)
+   {
+      // We expect at least one of these when conditons to eval to true, thus having the nothing directive utilized
+      #pragma omp metadirective \
+         when( device={kind(nohost)}: nothing ) \
+         when( device={arch("nvptx")}: nothing) \
+         when( implementation={vendor(amd)}: nothing ) \
+         default( parallel for map(tofrom: A) )
```

Comment:
```
test_metadirective_nothing.c:37:32: error: ‘map’ is not valid for ‘#pragma omp parallel for’
   37 |          default( parallel for map(tofrom: A) )
      |                                ^~~
```

Reply:
Thanks @tob2 ! Yes, the map needs to go with the enclosing target directive.